src/storage/hummock_sdk/src/table_watermark.rs

```diff
@@ -492,37 +492,12 @@ impl TableWatermarks {
         );
     }
 
-    pub fn clear_stale_epoch_watermark(&mut self, safe_epoch: u64) {
-        match self.watermarks.first() {
-            None => {
-                // return on empty watermark
-                return;
-            }
-            Some((earliest_epoch, _)) => {
-                if *earliest_epoch >= safe_epoch {
-                    // No stale epoch watermark needs to be cleared.
-                    return;
-                }
-            }
-        }
-        debug!("clear stale table watermark below epoch {}", safe_epoch);
-        let mut result_epoch_watermark = Vec::with_capacity(self.watermarks.len());
+    pub fn clear_stale_epoch_watermark(&mut self) {
```

Comment:
To reduce the affected logic in this PR, I think we can keep the table watermark logic unchanged, and only change to pass the `committed_epoch` as the safe epoch when calling this method. Same for `table_watermarks_by_table_ids_impl`.

Reply:
Fixed.

Reply:
I still modify `extract_single_table_watermark` to always return the watermark of the earliest epoch. I think it makes more sense than only returning when watermark's earliest epoch = committed epoch, because otherwise it's possible earliest epoch can never catch up with committed epoch.
cc @Li0k